desicochrane/basic-mailbox-api
# Add rest endpoints

**🤗commit-by-commit-friendly🤗**

# Why does this PR exist?

This adds the api endpoints to view a collection of messages and single messages.

### How should this be manually tested?

First update dependencies and make sure tests pass:

1. `composer update`
1. `phpunit`

Now you can import some messages from json and check the endpoints manually:

1. `php artisan mail-api:import <path-to-json-file>
1. with curl/postman/your favorite tool check out the index api view (remember to use basic auth with the admin token from your `.env` file)
    ```
    http://localhost/api/messages
    ```
1. Remember one of those ids and try view a single endpoint:
    ```
    http://localhost/api/messages/1
    ```

Now check that archived messages can be viewed:

1. in your db, set a time to the `archived_at` field of any number of messages
1. check `http://localhost/api/messages` and the archived message(s) should NOT be visible
1. check `http://localhost/api/messages?archived=true` and now you should ONLY see the archived message(s)

### Discussion

Very minimal api endpoints here, no filtering or sorting added. Also seems storage to not have users and permissions added yet.
